Lesson3/Task1/BankAccount.cs

```diff
@@ -0,0 +1,175 @@
+У метода два параметра: ссылка на объект класса банковский счет откуда снимаются деньги,
+второй параметр – сумма.
+*/
+
+public class BankAccount
+{
+    public enum BankAccountType
+    {
+        /// <summary>
+        /// Бюджетный счёт
+        /// </summary>
+        Budget,
+        /// <summary>
+        /// Валютный счёт
+        /// </summary>
+        Foreign_currency,
+        /// <summary>
+        /// Замороженный счёт
+        /// </summary>
+        Frozen,
+        /// <summary>
+        /// Застрахованный счёт
+        /// </summary>
+        Insured,
+        /// <summary>
+        /// Контокоррентный счёт
+        /// </summary>
+        Checking,
+        /// <summary>
+        /// Корреспондентский счёт
+        /// </summary>
+        Correspondent,
+        /// <summary>
+        /// Накопительный счёт
+        /// </summary>
+        Savings,
+        /// <summary>
+        /// Обезличенный металлический счёт
+        /// </summary>
+        Impersonal_metal,
+        /// <summary>
+        /// Общий счёт
+        /// </summary>
+        Total_score,
+        /// <summary>
+        /// Онкольный счёт
+        /// </summary>
+        On_call,
+        /// <summary>
+        /// Сводный счёт
+        /// </summary>
+        Consolidated,
+        /// <summary>
+        /// Ссудный счёт
+        /// </summary>
+        Loan,
+        /// <summary>
+        /// Текущий счёт
+        /// </summary>
+        Current,
+        /// <summary>
+        /// Транзитный счёт
+        /// </summary>
+        Transit,
+        /// <summary>
+        /// Фидуциарный счёт
+        /// </summary>
+        Fiduciary,
+        /// <summary>
+        /// Фондовый счёт
+        /// </summary>
+        Stock,
+        /// <summary>
+        /// Частный счёт
+        /// </summary>
+        Private,
+    }
+    const int NumberStart = 1000555;
+
+    /// <summary>
+    /// Переводит запрошенную сумму со счета [source]
+    /// </summary>
+    /// <param name="source">счет источник</param>
+    /// <param name="money"> сумма перевода</param>
+    /// <returns>true - операция успешна</returns>
+    public bool Transfer(BankAccount source, decimal money)
+    {
+        if (source == null) return false;
+        if (!source.GetMoney(money)) return false;
+        _Balance += money;
+        return true;
+    }
+
+
+
+    /// <summary>
+    /// конструктор по умолчанию
+    /// </summary>
+    public BankAccount()
+    {
+        SetNumber();
+        _Balance = 0;
+        _Type = BankAccountType.Savings;
+    }
+
+    public BankAccount(decimal Balance)
+    {
+        SetNumber();
+        _Balance = Balance;
+        _Type = BankAccountType.Savings;
+    }
+
+    public BankAccount(BankAccountType type)
+    {
+        SetNumber();
+        _Balance = 0;
+        _Type = type;
+    }
+
+    public BankAccount(decimal balance, BankAccountType type)
+    {
+        SetNumber();
+        _Balance = balance;
+        _Type = type;
+    }
```

Comment:
Это должен быть главный конструктор. А все остальные отсылать к нему.

```suggestion
    public BankAccount(decimal balance, BankAccountType type)
    {
        _Number = CreateNumber();
        _Balance = balance;
        _Type = type;
    }
```